test/e2e/virtual/apiexport/virtualworkspace_test.go

```diff
@@ -60,13 +62,15 @@ func TestAPIExportVirtualWorkspace(t *testing.T) {
 
 	cfg := server.DefaultConfig(t)
 
-	kcpClients, err := clientset.NewClusterForConfig(cfg)
+	clusterCfg := serverutil.NewClusterConfig(cfg)
```

Comment:
This is to make sure `clusterCfg` is scoped always, based on https://github.com/kcp-dev/kcp/pull/1376/files#r911498030

Reply:
Isn't every single caller of `server.DefaultConfig(t)` going to want this? Why not make it the default behavior? Or return both for the few cases where the "raw" one is needed as well for discovery-scoped clients? Or is there any ay to make the roundtripper (`NewClusterConfig()`) compatible with the host-prefix (`ConfigWithCluster()`)?

Reply:
We could explore moving this inside `server.DefaultConfig`. And maybe add another method on `server` to get the raw/unwrapped one.

We'll hopefully not need both routes when upstream Discovery supports a ctx

Reply:
Modified this to be other way around. `server.DefaultConfig` remains as is, and instead `server.ClusterConfig` gives us a wrapped one. Didn't want to modify the default config till we are sure that discovery calls will be scoped and artifact generation is fixed.